geoutils/georaster/raster.py

```diff
@@ -958,12 +1002,101 @@ def copy(self: RasterType, new_array: np.ndarray | None = None) -> RasterType:
 
         return cp
 
-    @property
-    def __array_interface__(self) -> dict[str, Any]:
-        if not self.is_loaded:
-            self.load()
+    def __array__(self) -> np.ndarray:
+        """Method to cast np.array() or np.asarray() function directly on Raster classes."""
+
+        return self._data
+
+    def __array_ufunc__(
+        self,
+        ufunc: Callable[[np.ndarray | tuple[np.ndarray, np.ndarray]], np.ndarray | tuple[np.ndarray, np.ndarray]],
+        method: str,
+        *inputs: Raster | tuple[Raster, Raster],
+        **kwargs: Any,
+    ) -> Raster | tuple[Raster, Raster]:
+        """
+        Method to cast NumPy universal functions directly on Raster classes, by passing to the masked array.
+        This function basically applies the ufunc (with its method and kwargs) to .data, and rebuilds the Raster from
+        self.__class__. The cases separate the number of input nin and output nout, to properly feed .data and return
+        Raster objects.
+        See more details in NumPy doc, e.g., https://numpy.org/doc/stable/user/basics.dispatch.html#basics-dispatch.
+        """
+
+        # If the universal function takes only one input
+        if ufunc.nin == 1:
+            # If the universal function has only one output
+            if ufunc.nout == 1:
+                return self.__class__(
+                    {
+                        "data": getattr(ufunc, method)(inputs[0].data, **kwargs),  # type: ignore
```

Comment:
Can't you define the type of the output (I assume `np.ndarray`) rather than ignoring mypy complain?

Reply:
The typing is already defined for `__array_ufunc__` and `__array_function__` in the function definition.

 Mypy still complains about something else specific to those lines, I think it's the indexing of `inputs` or something similar. I couldn't figure it out so I silenced it.